src/cpp/r/session/RSession.cpp

```diff
@@ -1494,14 +1494,17 @@ Error run(const ROptions& options, const RCallbacks& callbacks)
    bool loadInitFile = false;
    if (restartContext().hasSessionState())
    {
-      loadInitFile = restartContext().rProfileOnRestore();
+      loadInitFile = restartContext().rProfileOnRestore() && options.rProfileOnStart;
    }
    else
    {
-      loadInitFile = !s_suspendedSessionPath.exists()
-                     || options.rProfileOnResume
-                     || r::session::state::packratModeEnabled(
-                                                s_suspendedSessionPath);
+      // we run the .Rprofile if this is a brand new session and
+      // we are in a project and the ExecuteProfile setting is set, or we are not in a project
+      // alternatively, if we are resuming a session and the option is set to possibly run the .Rprofile
+      // we will only run it if the ExecuteProfile project setting is set (or we are not in a project)
+      loadInitFile = (!s_suspendedSessionPath.exists() && options.rProfileOnStart)
+                     || (options.rProfileOnResume && options.rProfileOnStart)
```

Comment:
I think making this control the Resume path as well is the right call, but I also think it's going to cause some confusion since it invisibly trumps a user preference, i.e.

true => follow UI pref on resume
false => never execute on resume, regardless of UI pref

For this reason I wonder if it's clearer to break the traditional UX taboo on double negatives and have the opt-in (checked) behavior be the override, i.e.

[ ] Do not run .Rprofile on session start or resume

I think it's clearer that this will trump the user pref. What do you think?